src/main/java/org/opensearch/security/dlic/rest/validation/RequestContentValidator.java

```diff
@@ -393,4 +543,227 @@ public String message() {
+                "Invalid " + fieldName + " [" + value + "]. Allowed values: " + String.join(", ", allowedValues)
+            );
+        }
+    }
+
+    /* ---------------------- JSON extraction helpers ---------------------- */
+
+    /**
+     * Extracts a required text field from JSON with length validation
+     */
+    public static String getRequiredText(JsonNode body, String fieldName, int maxLength) {
+        JsonNode node = body.get(fieldName);
+        if (node == null || node.isNull() || !node.isTextual()) {
+            throw new IllegalArgumentException("Field [" + fieldName + "] is required and must be a non-empty string");
+        }
+        String value = node.asText();
+        requireNonEmpty(fieldName, value);
+        validateMaxLength(fieldName, value, maxLength);
+        return value;
+    }
+
+    /**
+     * Extracts an optional text field from JSON with length validation
+     */
+    public static String getOptionalText(JsonNode body, String fieldName, int maxLength) {
+        JsonNode node = body.get(fieldName);
+        if (node == null || node.isNull()) {
+            return null;
+        }
+        if (!node.isTextual()) {
+            throw new IllegalArgumentException("Field [" + fieldName + "] must be a string when provided");
+        }
+        String value = node.asText();
+        if (value.isEmpty()) {
+            return null;
+        }
+        validateMaxLength(fieldName, value, maxLength);
+        return value;
+    }
+
+    /* ---------------------- specialized validators ---------------------- */
+
+    /**
+     * Validates a path-like value (no whitespace allowed)
+     */
+    public static void validatePath(String fieldName, String path, int maxLength) {
+        requireNonEmpty(fieldName, path);
+        validateMaxLength(fieldName, path, maxLength);
+        if (!path.equals(path.trim()) || path.chars().anyMatch(Character::isWhitespace)) {
+            throw new IllegalArgumentException(fieldName + " must not contain whitespace");
+        }
+    }
+
+    /**
+     * Validates a value against an allowed set
+     */
+    public static void validateFieldValueInSet(
+        String fieldName,
+        String value,
+        int maxLength,
+        Set<String> allowedValues,
+        String errorContext
+    ) {
+        requireNonEmpty(fieldName, value);
+        validateMaxLength(fieldName, value, maxLength);
+        if (allowedValues == null || allowedValues.isEmpty()) {
+            throw new IllegalStateException("No allowed " + errorContext + " configured");
+        }
+        if (!allowedValues.contains(value)) {
+            throw new IllegalArgumentException("Invalid " + fieldName + " [" + value + "]. Allowed " + errorContext + ": " + allowedValues);
+        }
+    }
+
+    /**
+     * Validates a JSON object node with non-empty string values
+     */
+    public static void validateObjectWithStringValues(String fieldName, JsonNode node) {
```

Comment:
Can we inline this if its just used in the single context of the `default_access_level` param? I think the name of this method is a bit of a misnomer if its simply validating that the key-value pair is non-empty.